modules/iam-role-for-service-accounts-eks/policies.tf

```diff
@@ -487,3 +488,254 @@ resource "aws_iam_role_policy_attachment" "karpenter_controller" {
   role       = aws_iam_role.this[0].name
   policy_arn = aws_iam_policy.karpenter_controller[0].arn
 }
+
+################################################################################
+# AWS Load Balancer Controller Policy
+################################################################################
+
+# https://github.com/kubernetes-sigs/aws-load-balancer-controller/blob/main/docs/install/iam_policy.json
+data "aws_iam_policy_document" "load_balancer_controller" {
+  count = var.create_role && var.attach_load_balancer_controller_policy ? 1 : 0
+
+  statement {
+    actions   = ["iam:CreateServiceLinkedRole"]
+    resources = ["*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "iam:AWSServiceName"
+      values   = ["elasticloadbalancing.${local.partition}"]
```

Comment:
Is it better to use [`aws_service` data source](https://github.com/hashicorp/terraform-provider-aws/pull/16640/files) instead of generating values manually? It is not in the documentation for some reason, yet. I am trying to think about a use case for it and whether we should be using it in some time?

Reply:
I see. We would need to use `dns_prefix` (`"amazonaws.com") but there is only `reverse_dns_prefix` (`"com.amazonaws"`) among attributes.